### PR TITLE
8333757: [lworld] sync javac code generation with the latest spec

### PR DESCRIPTION
**Pull request by vicente-romero-oracle**
Renaming the Preload attribute plus changing its contents. Side-effects, changes to class file API, javap,etc

---------
### Progress
- [x] Change must not contain extraneous whitespace

### Issue
 * [JDK-8333757](https://bugs.openjdk.org/browse/JDK-8333757): [lworld] sync javac code generation with the latest spec (**Bug** - P4)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/valhalla.git pull/1118/head:pull/1118` \
`$ git checkout pull/1118`

Update a local copy of the PR: \
`$ git checkout pull/1118` \
`$ git pull https://git.openjdk.org/valhalla.git pull/1118/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1118`

View PR using the GUI difftool: \
`$ git pr show -t 1118`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/valhalla/pull/1118.diff">https://git.openjdk.org/valhalla/pull/1118.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/valhalla/pull/1118#issuecomment-2155519117)

### CODE REVIEW

File: src/java.base/share/classes/java/lang/classfile/attribute/LoadableDescriptorsAttribute.java [hunk 1]
```diff
@@ -43,44 +44,28 @@
...
-        return of(Util.entryList(preloads));
-    }
-
-    /**
-     * {@return a {@code Preload} attribute}
-     * @param preloads the preload classes
-     */
-    static PreloadAttribute ofSymbols(ClassDesc... preloads) {
-        return ofSymbols(Arrays.asList(preloads));
+    static LoadableDescriptorsAttribute of(Utf8Entry... loadableDescriptors) {
```
Comment (liach) [inline src/java.base/share/classes/java/lang/classfile/attribute/LoadableDescriptorsAttribute.java:68]: I recommend another set of factory:
```java
static LoadableDescriptorsAttribute ofSymbols(ClassDesc... loadableDescriptors);
static LoadableDescriptorsAttribute ofSymbols(List<ClassDesc> loadableDescriptors);
```

Comment (vicente-romero-oracle) [reply src/java.base/share/classes/java/lang/classfile/attribute/LoadableDescriptorsAttribute.java:68]: I don't see the benefit tbh

Comment (vicente-romero-oracle) [reply src/java.base/share/classes/java/lang/classfile/attribute/LoadableDescriptorsAttribute.java:68]: we can always add it if necessary in the future

**Pull request closed without merging**